src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -28,6 +28,8 @@ internal BoundExpression BindQuery(QueryExpressionSyntax node, BindingDiagnostic
             var fromClause = node.FromClause;
             var boundFromExpression = BindLeftOfPotentialColorColorMemberAccess(fromClause.Expression, diagnostics);
 
+            ReportFieldOrValueContextualKeywordConflictIfAny(fromClause, fromClause.Identifier, diagnostics);
```

Comment:
>ReportFieldOrValueContextualKeywordConflictIfAny(fromClause, fromClause.Identifier, diagnostics); [](http://example.com/codeflow?start=12&length=97)

It looks like many of these checks, if not all, for query expressions could be moved into ```AddRangeVariable``` helper.